### PR TITLE
Upgrade all dependencies

### PR DESCRIPTION
**Pull request by robinandeer**
## What (highlights)

Upgrade ESLint v8.
Upgrade Husky v7, update setup.
Upgrade Node v16 (LTS), add nvmrc.

> I didn't upgrade to latest `lint-staged` since it's a pure ESM module which I couldn't get working with _husky_. We also don't really need any features not available in v11.

## Why

Keep things up to date.

### CODE REVIEW

File: codefresh.yml [hunk 1]
```diff
@@ -1,47 +1,47 @@
...
+    stage: 'clone'
 
   install:
-    title: "Install dependencies"
-    type: "freestyle"
-    image: "node:12.19.0"
-    working_directory: "${{clone}}"
+    title: 'Install dependencies'
+    type: 'freestyle'
+    image: 'node:lts-alpine'
```
Comment (robinandeer) [inline codefresh.yml:23]: This image will always track latest Node under Long Term Support.

**Review by guilhermespopolin — APPROVED** [review]
Just curious: What tutorial did you use to came up with this `husky` config? I saw that they've changed its inner works [here](https://blog.typicode.com/husky-git-hooks-javascript-config/) so did you just run the following?

`npx @husky/init` -> which probably has created the `.husky/` folder and has added the `prepare` script into `package.json`
`npx husky add .husky/pre-commit "npx lint-staged"` -> which has generated the `pre-commit` file

File: codefresh.yml (hunk 1 above)
Comment (guilhermespopolin) [reply codefresh.yml:23]: ℹ️

**Comment by robinandeer** [comment]
@guilhermespopolin yeah, I just followed the standard upgrade instructions that are printed to the console when you update the dependency I believe.

**Pull request merged**